c/driver_manager/adbc_driver_manager.cc

```diff
@@ -730,38 +720,41 @@ AdbcStatusCode AdbcLoadDriverFromInitFunc(AdbcDriverInitFunc init_func, size_t c
     return ADBC_STATUS_INTERNAL;                                               \
   }
 
-  auto result = init_func(count, driver, initialized, error);
+  auto result = init_func(version, raw_driver, error);
   if (result != ADBC_STATUS_OK) {
     return result;
   }
 
-  CHECK_REQUIRED(driver, DatabaseNew);
-  CHECK_REQUIRED(driver, DatabaseInit);
-  CHECK_REQUIRED(driver, DatabaseRelease);
-  FILL_DEFAULT(driver, DatabaseSetOption);
-
-  CHECK_REQUIRED(driver, ConnectionGetInfo);
-  CHECK_REQUIRED(driver, ConnectionNew);
-  CHECK_REQUIRED(driver, ConnectionInit);
-  CHECK_REQUIRED(driver, ConnectionRelease);
-  FILL_DEFAULT(driver, ConnectionCommit);
-  FILL_DEFAULT(driver, ConnectionGetObjects);
-  FILL_DEFAULT(driver, ConnectionGetTableSchema);
-  FILL_DEFAULT(driver, ConnectionGetTableTypes);
-  FILL_DEFAULT(driver, ConnectionReadPartition);
-  FILL_DEFAULT(driver, ConnectionRollback);
-  FILL_DEFAULT(driver, ConnectionSetOption);
-
-  FILL_DEFAULT(driver, StatementExecutePartitions);
-  CHECK_REQUIRED(driver, StatementExecuteQuery);
-  CHECK_REQUIRED(driver, StatementNew);
-  CHECK_REQUIRED(driver, StatementRelease);
-  FILL_DEFAULT(driver, StatementBind);
-  FILL_DEFAULT(driver, StatementGetParameterSchema);
-  FILL_DEFAULT(driver, StatementPrepare);
-  FILL_DEFAULT(driver, StatementSetOption);
-  FILL_DEFAULT(driver, StatementSetSqlQuery);
-  FILL_DEFAULT(driver, StatementSetSubstraitPlan);
+  if (version == ADBC_VERSION_1_0_0) {
+    auto* driver = reinterpret_cast<struct AdbcDriver*>(raw_driver);
+    CHECK_REQUIRED(driver, DatabaseNew);
+    CHECK_REQUIRED(driver, DatabaseInit);
+    CHECK_REQUIRED(driver, DatabaseRelease);
+    FILL_DEFAULT(driver, DatabaseSetOption);
+
+    CHECK_REQUIRED(driver, ConnectionGetInfo);
+    CHECK_REQUIRED(driver, ConnectionNew);
+    CHECK_REQUIRED(driver, ConnectionInit);
+    CHECK_REQUIRED(driver, ConnectionRelease);
+    FILL_DEFAULT(driver, ConnectionCommit);
+    FILL_DEFAULT(driver, ConnectionGetObjects);
+    FILL_DEFAULT(driver, ConnectionGetTableSchema);
+    FILL_DEFAULT(driver, ConnectionGetTableTypes);
+    FILL_DEFAULT(driver, ConnectionReadPartition);
+    FILL_DEFAULT(driver, ConnectionRollback);
+    FILL_DEFAULT(driver, ConnectionSetOption);
+
+    FILL_DEFAULT(driver, StatementExecutePartitions);
+    CHECK_REQUIRED(driver, StatementExecuteQuery);
+    CHECK_REQUIRED(driver, StatementNew);
+    CHECK_REQUIRED(driver, StatementRelease);
+    FILL_DEFAULT(driver, StatementBind);
+    FILL_DEFAULT(driver, StatementGetParameterSchema);
+    FILL_DEFAULT(driver, StatementPrepare);
+    FILL_DEFAULT(driver, StatementSetOption);
+    FILL_DEFAULT(driver, StatementSetSqlQuery);
+    FILL_DEFAULT(driver, StatementSetSubstraitPlan);
+  }
```

Comment:
Should we return a non-`ADBC_STATUS_OK` status if `version != ADBC_VERSION_1_0_0` here?